tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_type_test.bal

```diff
@@ -105,6 +106,22 @@ function testRegExpWithUserDefinedType() {
     assertEquality(re `AB*|[^c-d]{1,5}`, x7);
 }
 
+type T1 string:RegExp & readonly;
+type T2 regexp:RegExp & readonly;
+
+type Foo record {|
+    int e;
+    readonly regexp:RegExp f;
+|};
+
+Foo & readonly rf = {e: 1, f: re `test`};
+
+function testRegExpReadonlyLocalVars() {
+    string:RegExp & readonly _ = re `test`;
+    T1 & readonly _ = re `test`;
+    (T2 & readonly) & string:RegExp _ = re `test`;
+}
```

Comment:
Don't we need assertions here? Also kinda duplicating the other test, right?

Reply:
Updated in [d4f514](https://github.com/ballerina-platform/ballerina-lang/pull/41843/commits/d4f514b63dce28a4904ac3e21cc278607893631f).